### PR TITLE
Tacho : return value from team-factor

### PR DESCRIPTION
**Pull request by iyamazaki**
@trilinos/shylu 

## Motivation

To fix an error on H100.

## Testing

Tested on H100, V100, and Mi250, and also with OpenMP

### CODE REVIEW

File: packages/shylu/shylu_node/tacho/src/impl/Tacho_NumericTools_LevelSet.hpp [hunk 1]
```diff
@@ -4342,13 +4358,15 @@ class NumericToolsLevelSet : public NumericToolsBase<ValueType, DeviceType> {
...
-            Kokkos::fence();
+            Kokkos::deep_copy(h_rval, d_rval);
+            int rval = h_rval(0);
             if (rval != 0) {
               TACHO_TEST_FOR_EXCEPTION(rval, std::runtime_error, "GETRF (team) returns non-zero error code.");
             }
-            if (_status != 0) {
-              TACHO_TEST_FOR_EXCEPTION(rval, std::runtime_error, "GETRF (device) returns non-zero error code.");
-            }
+            //if (_status != 0) {
```
Comment (ndellingwood) [inline packages/shylu/shylu_node/tacho/src/impl/Tacho_NumericTools_LevelSet.hpp:4366]: Can be deleted in follow up cleanup?

File: packages/shylu/shylu_node/tacho/src/impl/Tacho_NumericTools_LevelSet.hpp [hunk 2]
```diff
@@ -3994,10 +4003,14 @@ class NumericToolsLevelSet : public NumericToolsBase<ValueType, DeviceType> {
               Kokkos::fence(); time_device += tick.seconds();
               tick.reset();
             }
-            Kokkos::fence();
+            Kokkos::deep_copy(h_rval, d_rval);
+            int rval = h_rval(0);
             if (rval != 0) {
               TACHO_TEST_FOR_EXCEPTION(rval, std::runtime_error, "SYTRF (team) returns non-zero error code.");
             }
+            //if (_status != 0) {
```
Comment (ndellingwood) [inline packages/shylu/shylu_node/tacho/src/impl/Tacho_NumericTools_LevelSet.hpp:4011]: Can be deleted in follow up cleanup?

File: packages/shylu/shylu_node/tacho/src/impl/Tacho_NumericTools_LevelSet.hpp [hunk 3]
```diff
@@ -2259,10 +2261,14 @@ class NumericToolsLevelSet : public NumericToolsBase<ValueType, DeviceType> {
               Kokkos::fence(); time_device += tick.seconds();
               tick.reset();
             }
-            Kokkos::fence();
+            Kokkos::deep_copy(h_rval, d_rval);
+            int rval = h_rval(0);
             if (rval != 0) {
               TACHO_TEST_FOR_EXCEPTION(rval, std::runtime_error, "POTRF (team) returns non-zero error code.");
             }
+            //if (_status != 0) {
```
Comment (ndellingwood) [inline packages/shylu/shylu_node/tacho/src/impl/Tacho_NumericTools_LevelSet.hpp:2269]: Can be deleted in follow up cleanup?

**Review by ndellingwood — APPROVED** [review]

**Comment by iyamazaki** [comment]
Thank you, @ndellingwood !!

**Pull request merged**